This notebook helps in understanding the purpose of the other two notebooks tags and links

In [5]:
import pandas as pd

In [6]:
tags = pd.read_csv(r"E:\end-to-end-collaborative-filtering-recommendation-system\data\tags.csv")
links = pd.read_csv(r"E:\end-to-end-collaborative-filtering-recommendation-system\data\links.csv")

Performing the Basic Data analysis for both the Tags and Links dataset

In [7]:
tags.head()

,userId,movieId,tag,timestamp
0,2,60756,funny,1445714994
1,2,60756,Highly quotable,1445714996
2,2,60756,will ferrell,1445714992
3,2,89774,Boxing story,1445715207
4,2,89774,MMA,1445715200


In [8]:
tags.tail()

,userId,movieId,tag,timestamp
3678,606,7382,for katie,1171234019
3679,606,7936,austere,1173392334
3680,610,3265,gun fu,1493843984
3681,610,3265,heroic bloodshed,1493843978
3682,610,168248,Heroic Bloodshed,1493844270


In [9]:
tags.columns

Index(['userId', 'movieId', 'tag', 'timestamp'], dtype='object')

In [10]:
tags.isnull().sum()

userId       0
movieId      0
tag          0
timestamp    0
dtype: int64

In [11]:
tags.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3683 entries, 0 to 3682
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   userId     3683 non-null   int64 
 1   movieId    3683 non-null   int64 
 2   tag        3683 non-null   object
 3   timestamp  3683 non-null   int64 
dtypes: int64(3), object(1)
memory usage: 115.2+ KB


In [12]:
tags.shape

(3683, 4)

In [13]:
#tags["movie"]

In [14]:
tags["movieId"].nunique()

1572

In [15]:
tags["userId"].count()

np.int64(3683)

In [16]:
tags["userId"].nunique()

58

In [17]:
tags["userId"].duplicated().sum()

np.int64(3625)

Conclusion of TAGS Dataset

1. THis dataset contains columns userId, MovieId, tag, timestamp
2. No missing values
3. Shape :- (3683, 4)
4. Out of 3683 userIds only 58 are unique and in movies 1572
5. Duplicate UserId sum 3625 

Similarly for links

In [18]:
links.head()

,movieId,imdbId,tmdbId
0,1,114709,862.0
1,2,113497,8844.0
2,3,113228,15602.0
3,4,114885,31357.0
4,5,113041,11862.0


In [19]:
links.shape

(9742, 3)

In [20]:
links.columns

Index(['movieId', 'imdbId', 'tmdbId'], dtype='object')

In [21]:
links.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9742 entries, 0 to 9741
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   movieId  9742 non-null   int64  
 1   imdbId   9742 non-null   int64  
 2   tmdbId   9734 non-null   float64
dtypes: float64(1), int64(2)
memory usage: 228.5 KB


In [22]:
links.isnull().sum()

movieId    0
imdbId     0
tmdbId     8
dtype: int64

In [23]:
links.duplicated().sum()

np.int64(0)